org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/refactoring/IRenameStrategy.xtend

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2017 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ide.refactoring
+
+import com.google.inject.ImplementedBy
+import com.google.inject.Inject
+import org.eclipse.emf.ecore.EAttribute
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.EcorePackage
+import org.eclipse.xtext.ide.refactoring.RefactoringIssueAcceptor.Severity
+import org.eclipse.xtext.resource.IResourceServiceProvider
+import org.eclipse.xtext.ide.serializer.IChangeSerializer
+
+/**
+ * Called to rename an element in the {@link IChangeSerializer} based refactoring.
+ * 
+ * Clients may extend the {@link DefaultImpl} to customize the behavior or implement 
+ * this interface directly.
+ * 
+ * Changes are usually performed in the The {@link RenameContext} 
+ * 
+ * @author koehnlein - Initial contribution and API
+ * @since 2.13
+ */
+@ImplementedBy(IRenameStrategy.DefaultImpl)
+interface IRenameStrategy {
+
+	def void applyRename(RenameContext context)
+	
+	def String getCurrentName(EObject element)
```

Comment:
Not sure how `getCurrentName(EObject element)` makes sense. 
* The user triggers rename refactoring on a token, not on an EObject
* EObjects can have multiple names at the same time, i.e. multiple renamable EAttributes.
* Since rename refactoring is triggered on a token, the token's value is the "current name", isn't it?

Maybe what we need here is method such as 
```
boolean canRename(ISemanticRegion token)
```
to check rename refactoring can be triggered on the current token?